src/qz/installer/certificate/firefox/locator/WindowsAppLocator.java

```diff
@@ -49,7 +57,67 @@ public WindowsAppLocator(String name, String path, String version) {
         return appList;
     }
 
-    public static AppLocator getAppInfo(String name, String key, String suffix) {
+    @Override
+    public ArrayList<String> getPids(boolean parentPids, ArrayList<String> processNames) {
+        ArrayList<String> pidList = new ArrayList<>();
+        ArrayList<String> parentPIDList = new ArrayList<>();
+
+        if (processNames.isEmpty()) return pidList;
+
+        WIN32_PID_QUERY[WIN32_PID_QUERY_INPUT_INDEX] = "(Name='" + String.join("' OR Name='", processNames) + "')";
+        String[] response = ShellUtilities.executeRaw(WIN32_PID_QUERY).split("[\\r\\n]+");
+
+        // Skip the first result (the first row is column headers)
+        for (int i = 1; i < response.length; i++) {
+            String[] row =  response[i].split("[\\s,]+");
+
+            parentPIDList.add(row[0]);
+            pidList.add(row[1]);
+        }
+
+        if(parentPids) {
+            // Remove all processes that are child to another process in this set
+            for(int i = pidList.size() - 1; i >= 0; i--) {
```

Comment:
Can't we replace this whole loop with just `pidList.removeAll(parentPIDList);`?

Reply:
I thought so too, the short answer is no.  It's a strange parent/orphan relationship since they're pairs.  It breaks if that's done.

Reply:
Decided to refactor `parentPids` out completely, it's no longer used.